src/test/scala/org/bitcoins/core/crypto/TransactionSignatureCreatorSpec.scala

```diff
@@ -37,16 +38,62 @@ class TransactionSignatureCreatorSpec extends Properties("TransactionSignatureCr
 
   property("generate a valid signature for a p2sh transaction") =
-    Prop.forAll(TransactionGenerators.p2SHTransaction) {
+    Prop.forAll(TransactionGenerators.signedP2SHTransaction) {
       case (txSignatureComponent: TransactionSignatureComponent, _) =>
         //run it through the interpreter
         val program = ScriptProgram(txSignatureComponent)
         val result = ScriptInterpreter.run(program)
         //can be ScriptErrorPushSize if the redeemScript is larger than 520 bytes
         Seq(ScriptOk, ScriptErrorPushSize).contains(result)
     }
+
+  property("generate a valid signature for a valid and spendable cltv transaction") =
+    Prop.forAllNoShrink(TransactionGenerators.spendableCLTVTransaction :| "cltv_spendable") {
+      case (txSignatureComponent: TransactionSignatureComponent, _, scriptNumber) =>
+        //run it through the interpreter
+        require(txSignatureComponent.transaction.lockTime.underlying >= scriptNumber.underlying, "TxLocktime must be satisifed so it should be greater than or equal to  " +
+          "the cltv value. Got TxLockTime : " + txSignatureComponent.transaction.lockTime.underlying + " , and cltv Value: " +
+          scriptNumber.underlying)
+        val program = ScriptProgram(txSignatureComponent)
+        val result = ScriptInterpreter.run(program)
+        val cltv = CLTVScriptPubKey(txSignatureComponent.scriptPubKey.hex)
+        Seq(ScriptOk).contains(result)
+    }
+
+  property("generate a valid signature for a validly constructed, but NOT spendable cltv transaction") =
+    Prop.forAllNoShrink(TransactionGenerators.unspendableCLTVTransaction :| "cltv_unspendable") {
+      case (txSignatureComponent: TransactionSignatureComponent, _, scriptNumber) =>
+        //run it through the interpreter
+        require(txSignatureComponent.transaction.lockTime.underlying < scriptNumber.underlying, "TxLocktime must not be satisifed so it should be less than " +
+          "the cltv value. Got TxLockTime : " + txSignatureComponent.transaction.lockTime.underlying + " , and cltv Value: " +
+          scriptNumber.underlying)
+        val program = ScriptProgram(txSignatureComponent)
+        val result = ScriptInterpreter.run(program)
+        val cltv = CLTVScriptPubKey(txSignatureComponent.scriptPubKey.hex)
+        Seq(ScriptErrorUnsatisfiedLocktime, ScriptErrorPushSize).contains(result)
+    }
+
+  property("generate a valid signature for a valid and spendable csv transaction") =
+    Prop.forAllNoShrink(TransactionGenerators.spendableCSVTransaction :| "spendable csv") {
+      case (txSignatureComponent: TransactionSignatureComponent, keys, scriptNumber, sequence) =>
+        //run it through the interpreter
+        val program = ScriptProgram(txSignatureComponent)
+        val result = ScriptInterpreter.run(program)
+        Seq(ScriptOk, ScriptErrorPushSize).contains(result)
+    }
+
+  property("generate a valid signature for a validly constructed but unspendable csv transaction") =
```

Comment:
We don't get to check the signature in this case, so I would probably just name this property `fail to verify a transaction with a csv value that hasn't passed yet`